src/front/spv.rs

```diff
@@ -330,33 +365,46 @@ struct LookupSampledImage {
     image: Handle<crate::Expression>,
     sampler: Handle<crate::Expression>,
 }
-
 struct DeferredFunctionCall {
     source_handle: Handle<crate::Function>,
     expr_handle: Handle<crate::Expression>,
     dst_id: spirv::Word,
 }
-
-enum Terminator {
+pub enum Merge {
```

Comment:
Nice enum!

Where are we matching it to anything? If we have to extract the block id somewhere, and we don't expect the enum to grow, we can refactor it into a struct:
```rust
pub enum Merge {
  block_id: spirv::Word,
  continue_id: Option<spirv::Word>,
}
```